lib/sqlalchemy/sql/elements.py

```diff
@@ -2705,7 +2755,10 @@ def append(self, clause):
     def _from_objects(self) -> List[FromClause]:
         return list(itertools.chain(*[c._from_objects for c in self.clauses]))
 
-    def self_group(self, against=None):
+    def self_group(
+        self, against: Optional[OperatorType] = None
+    ) -> ClauseElement:
+        assert against is not None
```

Comment:
Some of the `self_group()` overrides require `against` argument, which conflicts with `Optional` annotation - used `assert` to narrow that down without changing the actual signature.

Reply:
I think it would be best to do something like `if against is not None and self.group and operators.is_precedent(self.operator, against):`, because otherwise a call to `self_group` with `against` of `None` fails but is not type-checked.

https://github.com/GodelTech/sqlalchemy/blob/c6893624a6e63b4e532ce5f8b6f5e86e29abab44/lib/sqlalchemy/sql/elements.py#L2758-L2765